packages/kbn-i18n/GUIDELINE.md

```diff
@@ -0,0 +1,330 @@
+  </select>
+  ```
+
+- for placeholder:
+
+  ```js
+  <EuiFieldText
+      name="indexPatternId"
+      placeholder={intl.formatMessage({
+        id: 'kbn.management.createIndexPattern.stepTime.options.patternPlaceholder',
+        defaultMessage: 'custom-index-pattern-id' })}
+  />
+  ```
+
+- for `aria-label` attribute and tooltip
+
+  ```js
+  <button
+      aria-label="{{'kbn.management.editIndexPattern.removeAria' | i18n: {defaultMessage: 'Remove index pattern'} }}"
+      tooltip="{{'kbn.management.editIndexPattern.removeTooltip' | i18n: {defaultMessage: 'Remove index pattern'} }}"
+      >
+  </button>
+  ```
+
+- for errorMessage:
+
+  ```js
+  errors.push(
+      intl.formatMessage(
+              {
+                  id: 'kbn.management.createIndexPattern.step.invalidCharactersErrorMessage',
+                  defaultMessage: 'An index pattern cannot contain spaces or the characters: {characterList}'
+              },
+              { characterList }
+      ));
+  ```
+
+- for toggleSwitch
+
+  ```js
+  <EuiSwitch
+      label={<FormattedMessage
+        id="kbn.management.createIndexPattern.includeSystemIndicesToggleSwitch"
+        defaultMessage="Include system indices"
+      />}
+  />
+  ```
+
+
+### Text with plurals
+
+The numeric input is mapped to a plural category, some subset of "zero", "one", "two", "few", "many", and "other" depending on the locale and the type of plural. There are languages with multiple plural forms [Language Plural Rules](http://www.unicode.org/cldr/charts/latest/supplemental/language_plural_rules.html).
+
+Here is an example of message translation depending on a plural category:
+
+```js
+<span i18n-id="kbn.management.editIndexPattern.mappingConflictLabel"
+      i18n-default-message="{conflictFieldsLength, plural, one {A field is} other {# fields are}} defined as several types (string, integer, etc) across the indices that match this pattern."
+      i18n-values="{ conflictFieldsLength: conflictFields.length }"></span>
+```
+
+When `conflictFieldsLength` equals 1, the result string will be `"A field is defined as several types (string, integer, etc) across the indices that match this pattern."`. In cases when `conflictFieldsLength` has value of 2 or more, the result string - `"2 fields are defined as several types (string, integer, etc) across the indices that match this pattern."`.
+
+### Splitting
+
+Splitting sentences into several keys often inadvertently presumes a grammar, a sentence structure, and such composite strings are often very difficult to translate.
+
+- Do not divide a single sentence into different labels unless you have absolutely no other choice.
+- Do not divide sentences that belong together into separate labels.
+
+  For example:
+
+  `The following dialogue box indicates progress. You can close it and the process will continue to run in the background.`
+
+  If this sentence is separated it’s possible that the context of the `'it'` in `'close it'` will be lost.
```

Comment:
I wonder if there's a better word to use here than "sentence" to describe a group. How about paragraph? Or "sentence or group of sentences"? Since the example there is actually two sentences.